gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractJPackageTask.kt

```diff
@@ -368,6 +378,31 @@ abstract class AbstractJPackageTask @Inject constructor(
             cliArg("--license-file", licenseFile)
             cliArg("--resource-dir", jpackageResources)
 
+            val propertyFilesDirJava = propertyFilesDir.ioFile
+            fileOperations.clearDirs(propertyFilesDir)
+
+            val fileAssociationFiles = fileAssociations.orNull.orEmpty()
+                .groupBy { it.extension }
+                .mapValues { (extension, associations) ->
+                    associations.mapIndexed { index, association ->
+                        propertyFilesDirJava.resolve("FA${extension}${if (index > 0) index.toString() else ""}.properties")
+                            .apply {
+                                writeText(
+                                    """
+                                        mime-type=${association.mimeType}
+                                        extension=${association.extension}
+                                        description=${association.description}
```

Comment:
```suggestion
                                        description=${association.description}
                                        icon=${association.iconPath}
```

Let's also add support for settings icons (or create a follow-up issue for adding it).

Reply:
As icons are platform-specific, how would you like to see the api: writing a single `fileAssociation` function with `windowsIcon` etc. or writing `fileAssociation` function with `iconFile` for every platform?

Reply:
As suggested before, we need to move it to `AbstractPlatformSettings`. This way, you should specify `fileAssociation` for each platform separately, or extract own function, in which you can apply own logic of choosing the icon.

See similar property `AbstractPlatformSettings.iconFile`